### PR TITLE
add scoring and sort to search plugin

### PR DESCRIPTION
**Pull request by spion**
Updates the search plugin and adds scoring to search results. Search results that contain multiple keywords are scored higher than those that contain only one and appear first on the result list.

Headings add score +3, content matches +2. How will that work?
* For two keywords, a heading containing only one will be scored lower than body text containing both. But if a heading contains both, it will score higher than body text containing both.
* For 3 keywords, a heading containing 2 of the 3 will score the same as body text containing all 3
* For 4 keywords, a heading containing 3 of the 4 will score higher than body text containing all 4

------

Before:
image

------

After:
image

### CODE REVIEW

**Comment by spion** [comment]
p.s. we could also use tf-idf (term frequency inverse document frequency) for the scoring scheme, so that instead of every word counting the same, they have a score

`numberOfOccurancesInDoc / numberOfDocsWhereItOccurs`

**Comment by timaschew** [comment]
Please run these commands to trigger netlify deployment for this Pull Request:
```sh
git commit --amend --no-edit
git push --force
```

**Comment by spion** [comment]
done

**Comment by spion** [comment]
bump?

File: src/plugins/search/search.js [hunk 1]
```diff
@@ -96,29 +96,27 @@ export function search(query) {
...
     const postContent = post.body && post.body.trim()
     const postUrl = post.slug || ''
 
-    if (postTitle && postContent) {
-      keywords.forEach(keyword => {
+    if (postTitle) {
+      keywords.forEach( keyword => {
         // From https://github.com/sindresorhus/escape-string-regexp
-        const regEx = new RegExp(
+        let regEx = new RegExp(
```
Comment (timaschew) [inline src/plugins/search/search.js]: Why you changed it to `let`?

Comment (spion) [reply src/plugins/search/search.js]: No reason, I don't personally buy the modern const vs let usage so my usual let usage slipped up in this code. Changing.

Comment (timaschew) [reply src/plugins/search/search.js]: >> edit: done. Should I squash the commit?

Yes, please.

Comment (spion) [reply src/plugins/search/search.js]: done

**Pull request merged**